tests/ut/test_simd.cc

```diff
@@ -113,3 +134,88 @@ TEST_CASE("Test PQ Search SIMD", "[pq]") {
         }
     }
 }
+
+TEST_CASE("Test fp16 distance", "[fp16]") {
+    using Catch::Approx;
+    auto dim = GENERATE(as<size_t>{}, 1, 2, 10, 69, 128, 141, 510, 1024);
```

Comment:
I'd test `1,2,4,5,10,13,21,29` as well
Testing high dimensionalities, larger than 96 or 128 does not make sense to me, because the code contains dimensionality granularities and if-then-else / while constructions for like values 32, 16, 8, 4.